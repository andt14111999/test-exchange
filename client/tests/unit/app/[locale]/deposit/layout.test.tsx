import React from "react";
import { render } from "@testing-library/react";
import DepositLayout from "@/app/[locale]/deposit/layout";

// Mock ProtectedLayout component
jest.mock("@/components/protected-layout", () => ({
  ProtectedLayout: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="protected-layout">{children}</div>
  ),
}));

describe("DepositLayout", () => {
  const mockChildren = <div data-testid="test-children">Test Content</div>;

  it("should render the layout with children", () => {
    const { getByTestId } = render(
      <DepositLayout>{mockChildren}</DepositLayout>,
    );

    // Check if ProtectedLayout is rendered
    expect(getByTestId("protected-layout")).toBeInTheDocument();

    // Check if children are rendered
    expect(getByTestId("test-children")).toBeInTheDocument();
  });

  it("should have correct styling classes", () => {
    const { container } = render(<DepositLayout>{mockChildren}</DepositLayout>);

    const mainDiv = container.firstChild?.firstChild as HTMLElement;
    expect(mainDiv).toHaveClass(
      "min-h-[calc(100vh-theme(spacing.14))]",
      "bg-gradient-to-b",
      "from-background",
      "to-muted",
      "mx-auto",
      "max-w-screen-lg",
    );
  });

  it("should match snapshot", () => {
    const { container } = render(<DepositLayout>{mockChildren}</DepositLayout>);
    expect(container).toMatchSnapshot();
  });
});
