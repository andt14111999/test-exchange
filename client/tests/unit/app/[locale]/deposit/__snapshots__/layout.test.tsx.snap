// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositLayout should match snapshot 1`] = `
<div>
  <div
    data-testid="protected-layout"
  >
    <div
      class="min-h-[calc(100vh-theme(spacing.14))] bg-gradient-to-b from-background to-muted mx-auto max-w-screen-lg"
    >
      <div
        data-testid="test-children"
      >
        Test Content
      </div>
    </div>
  </div>
</div>
`;
